parser.go

```diff
@@ -304,6 +304,12 @@ func (l *DecodingLayerParser) DecodeLayers(data []byte, decoded *[]LayerType) (e
 	if !l.IgnorePanic {
 		defer panicToError(&err)
 	}
+	if len(data) == 0 {
+		// For compatibility with pre 1.1.18 behavior where an empty
+		// slice didn't cause an error. See
+		// https://github.com/google/gopacket/issues/846
```

Comment:
I think we have to do `*decoded = (*decoded)[:0]`.

Reply:
Agree, thanks for spotting.